### ISSUE TITLE
Give error/warning when using set(...) on nonexistent property

### ISSUE THREAD

**Issue by Error7Studios** [godotengine/godot-proposals#1494]
**Describe the project you are working on:**
(Applies to any project)

**Describe the problem or limitation you are having in your project:**
No error/warning is given when using `set(...)` on a nonexistent property
I imagine most people would want to get an error/warning instead of no feedback at all.
They probably made a typo when writing the property name, or wrote the wrong word, like `visibility` instead of `visible`.

**Describe the feature / enhancement and how it helps to overcome the problem or limitation:**
Give an error when a user tries to do this.
(Or a warning, at the very least)

**Describe how your proposal will work, with code, pseudocode, mockups, and/or diagrams:**
![SetPropertyErrorMessage](https://user-images.githubusercontent.com/60025044/92985517-f3195d00-f478-11ea-88df-b71cfd2b0a8c.png)

**If this enhancement will not be used often, can it be worked around with a few lines of script?:**
No.

**Is there a reason why this should be core and not an add-on in the asset library?:**
This is a core feature.

**Comment by cgbeutler** [comment]
This change may cause strange side effects. I'm pretty sure that `set` is currently used in the loading of objects from a file. That means that changing exported properties in a script, then loading that obj from the file would result in a blasting of errors for the exports removed or renamed. One way around this would be to maybe make the warning an option on set? Like `void set ( String property, Variant value, Bool warn_on_failure )` or something. Another option would be to return a bool from `set()`, like `_set()` has. This could be named differently like `try_set()`?
Anyway, food for thought.

**Comment by Calinou** [comment]
@cgbeutler Indeed. When I last tried making `Object.set()` print a message when the property didn't exist, I got a lot of errors when opening the editor.

**Comment by Error7Studios** [comment]
Any of those proposed new functions sound good to me.

**Comment by cgbeutler** [comment]
I know some other codebases usually use the convention `get_or_default(property, backup=null)`. It may be good to just rework both get and set to a similar format? I, personally, would like something like:
```
# Warnings
get(property)
set(property)
# No Warnings
get_or_default(property, backup=null)
set_or_ignore(property, value)
```

**Comment by zorbathut** [comment]
I've been trying this on my local code and so far it's working great. This is a *very* primitive approach, but I've got my current snippet of code up on [a branch](https://github.com/godotengine/godot/compare/master...zorbathut:godot:pr_propwarning), which I'm happy to set up as a code review but thought I'd write it up here first.

----

Now, some notes. The initial motivation for this was when I spent like two hours debugging an animation problem that turned out to be a variable that was read-only. This code doesn't do a good job of dealing with that error because it prints a generic "failed to set property" message (which would still have been a lot more useful than a silent failure!) I kinda wish it printed something more explanatory - "failed to set property because it's a read-only animation property" - but there's a few choices here.

* Print *two* error messages in that case, one detailed and one not-detailed.
* Complicate the interface considerably by virtue of needing to pass error messages around or suppress upstream error messages.
* Just print a single general error message.

I grudgingly chose the third, and so far that's been a better choice than the others. But the other choices are possible!

----

Objections:

"Won't this spam a lot of errors if you open a level with missing properties?"

*I want it to do this*. This sounds like a *desirable* feature!

"Should we add a get_or_default() function to suppress errors?"

We could. I'm personally not sure I'd use that often. I'd be more likely to check to see if a property exists, *then* call get(). But it's not a terrible idea, it's just not something I particularly needed.

"Can we do something to pre-emptively warn the user in GDScript?"

Probably, but I don't use GDScript. Also not a bad idea! But not part of this change.

----

I think having this has already saved me time, given that it took little time to write and has helped me with one bug so far. I think it's a significant overall benefit and should be made part of the main codebase. I'm also happy to make tweaks as necessary.

I'll post this as an actual code review if given a thumbs-up by a main contributor, or after a week, whichever comes first.

**Comment by cgbeutler** [comment]
My comment about spamming errors was more for future implementers. This change is deceptively larger than it seems, as 'set' is used in a ton of places in the engine that expect current behavior. I think this should be implemented, but I don't see anyone taking on such a massive task for such a small number of up votes.

**Comment by zorbathut** [comment]
Well, I'm happy to do that, because I think the safety provided by good error reporting is worth it. Do you know offhand of any places that expect to `set()` properties that don't exist?

**Comment by Calinou** [comment]
Regarding internal engine usage, what we can do is make the engine's internal `set()` property act like `set_or_null()`, and create a new method that is bound to scripting that prints errors on unknown properties (plus `set_or_null()`).

**Comment by zorbathut** [comment]
That's kinda the approach I'm taking here; as long as we're not looking for detailed error messages, the internal `_set()` function is already doing that, and it's just the external-facing `set()` function that actually prints the error message.

Also, I totally forgot about this, but the `set()`/`get()` functions take a `bool*` to optionally return success. The code I've got suppresses the error message if you've provided a pointer to fill.
